### ISSUE TITLE
iOS-Crash

### ISSUE THREAD

**Issue by shenyinghao** [getsentry/sentry-cocoa#2752]
### Platform

iOS

### Installed

CocoaPods

### Version

8.1.0

### Steps to Reproduce

已崩溃：com.apple.main-thread
SIGSEGV 0x0000000000000008
0
libsystem_blocks.dylib
_Block_copy + 56
5
libdispatch.dylib
dispatch_after + 280
6
CIDER
SentryDispatchQueueWrapper.m - 第 68 行
-[SentryDispatchQueueWrapper dispatchAfter:block:] + 68
7
CIDER
SentryTracer.m - 第 223 行
-[SentryTracer dispatchIdleTimeout] + 223
8
CIDER
SentryTracer.m - 第 344 行
-[SentryTracer spanFinished:] + 344
9
CIDER
SentrySpan.m - 第 149 行
-[SentrySpan finishWithStatus:] + 149
10
CIDER
SentryNSDataTracker.m - 第 257 行
-[SentryNSDataTracker finishTrackingNSData:span:] + 257
11
CIDER
SentryNSDataTracker.m - 第 117 行
-[SentryNSDataTracker measureNSDataFromFile:options:error:method:] + 117
12
CIDER
SentryNSDataSwizzling.m - 第 100 行
__38+[SentryNSDataSwizzling swizzleNSData]_block_invoke_2.43 + 100
13
Foundation
+[NSData(NSData) dataWithContentsOfFile:options:error:] + 52
40
libswiftUIKit.dylib
UIApplicationMain(_:_:_:_:) + 104

### Expected Result

no crash

### Actual Result

a lot of crash

**Comment by shenyinghao** [comment]
8.1.0版本造成的Crash太多了，什么时间能有个稳定版本；

**Comment by shenyinghao** [comment]
8.1.0版本造成的Crash太多了，什么时间能有个稳定版本；

**Comment by brustolin** [comment]
I believe this was fixed in https://github.com/getsentry/sentry-cocoa/pull/2697.
Can you try the new version 8.2.0?

**Comment by shenyinghao** [comment]
> 

I'll try, thank you

**Comment by Fwang36** [comment]
Hey there, 

Closing this ticket for now.  But please feel free to reopen this if the issue persists.

Thanks!

**Comment by shenyinghao** [comment]
> 

8.2.0 always crash

**Comment by philipphofmann** [comment]
@shenyinghao, what do you exactly mean by ?

> 8.2.0 always crash

**Comment by mshibanami** [comment]
I have encountered similar exceptions in our apps while using version 8.2.0 of this SDK, as follows:

![Screenshot 2023-04-21 at 11 50 00 am](https://user-images.githubusercontent.com/1333214/233522272-c8022374-15ae-4db7-b3e1-1c9edab43603.jpg)

The crash itself is not always, but whenever it happens, it appears to be triggered by this code in our `Preferences.cache<T>()` static method:

```swift
        let data = NSKeyedArchiver.archivedData(withRootObject: value as Any)
        try data.write(to: pathForKey(slugifiedKey))
```

Unfortunately, this reduces 1 ~ 1.5% of the crash-free rate for our app since we started using Sentry this week, so for now, I'll turn off the swizzling feature by following [this](https://docs.sentry.io/platforms/apple/guides/ios/configuration/swizzling/).

Sorry @philipphofmann @Fwang36, would you please have another look?
Please let me know if you need any additional information. Thank you for your help.

**Comment by brustolin** [comment]
Hello @mshibanami. We had a few updates since 8.2.0 to solve this problems. 
I want to encourage you to try our newest versions.

**Comment by mshibanami** [comment]
@brustolin Thanks for the information. I'll try it.

**Comment by mshibanami** [comment]
@brustolin Sorry for providing incorrect information. We actually encountered this issue while using version 8.4.0.
Do you think there's a chance that the issue will be resolved in version 8.5.0?

**Comment by brustolin** [comment]
8.5 is not meant for this problem in particular, so we would need to investigate whats going on.

Can you please email me the name of your org and project so I can look it up?
dhiogo.brustolin@sentry.io
